packages/core/src/State.ts

```diff
@@ -39,7 +41,8 @@ interface MachineSnapshotBase<
   TChildren extends Record<string, AnyActorRef | undefined>,
   TTag extends string,
   TOutput,
-  TResolvedTypesMeta = TypegenDisabled
+  TResolvedTypesMeta = TypegenDisabled,
+  TConfig extends StateConfig2 = any
```

Comment:
this shouldn't receive the whole `TConfig` but rather the `TStateValue` - in other words, we should precompute this and pass the resolved value here